samples/bluetooth/peripheral_hr_coded/src/main.c

```diff
@@ -0,0 +1,199 @@
+#include <zephyr.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/services/bas.h>
+#include <bluetooth/services/hrs.h>
+
+static struct k_work start_advertising_worker;
+
+static struct bt_le_ext_adv *adv;
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_UUID16_ALL, 0x0d, 0x18, 0x0f, 0x18, 0x0a, 0x18),
+};
+
+static void connected(struct bt_conn *conn, u8_t conn_err)
+{
+	if (conn_err) {
+		printk("Connection failed (err %d)\n", conn_err);
+		return;
+	}
+	
+	default_conn = bt_conn_ref(conn);
+	printk("Connected\n");
+}
+
+static void disconnected(struct bt_conn *conn, u8_t reason)
+{
+	printk("Disconnected (reason 0x%02x)\n", reason);
+
+	if (default_conn) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+	}
+
+	k_work_submit(&start_advertising_worker);
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static int create_advertising_coded(void)
+{
+	int err;
+	uint8_t adv_index;
+	struct bt_le_adv_param param = {};
+
+	param.options  = BT_LE_ADV_OPT_CONNECTABLE;
+	param.options |= BT_LE_ADV_OPT_EXT_ADV;
+	param.options |= BT_LE_ADV_OPT_CODED;
+	param.id = BT_ID_DEFAULT;
+	param.sid = 0;
+	param.interval_min = BT_GAP_ADV_FAST_INT_MIN_2;
+	param.interval_max = BT_GAP_ADV_FAST_INT_MAX_2;
+
+	err = bt_le_ext_adv_create(&param, NULL, &adv);
+	if (err) {
+		printk("Failed to create advertiser set (%d)\n", err);
+		return -ENOEXEC;
+	}
+
+	adv_index = bt_le_ext_adv_get_index(adv);
+
+	printk("Created adv id: %d, adv: %p\n", adv_index, adv);
+
+	err = bt_le_ext_adv_set_data(adv, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err) {
+		printk("Failed to set advertising data (%d)\n", err);
+	}
+
+	return 0;
+}
+
+static void start_advertising_coded(struct k_work *item)
+{
+	int err;
+	uint8_t adv_index;
+
+    struct bt_le_ext_adv_start_param start_param = {
+		.timeout    = 0,
+		.num_events = 0
+	};
+
+	err = bt_le_ext_adv_start(adv, &start_param);
+	if (err) {
+		printk("Failed to start advertising set (%d)\n", err);
+		return;
+	}
+
+	adv_index = bt_le_ext_adv_get_index(adv);
+
+	printk("Advertiser[%d] %p set started\n", adv_index, adv);
+}
+
+static void bt_ready(void)
+{
+	int err = 0;
+
+	printk("Bluetooth initialized\n");
+
+	k_work_init(&start_advertising_worker, start_advertising_coded);
+
+	err = create_advertising_coded();
+	if (err) {
+		printk("Advertising failed to create (err %d)\n", err);
+		return;
+	}
+
+	k_work_submit(&start_advertising_worker);
+}
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing cancelled: %s\n", addr);
+}
+
+static struct bt_conn_auth_cb auth_cb_display = {
+	.cancel = auth_cancel,
```

Comment:
Well, this is how it is implemented in the Zephyr [`peripheral_hr`](https://github.com/zephyrproject-rtos/zephyr/blob/zephyr-v2.3.0/samples/bluetooth/peripheral_hr/src/main.c) sample.
If you are fine with it, I will remove it..

Reply:
I see. Best thing would be to align it with the rest of the NCS samples, like this:
https://github.com/nrfconnect/sdk-nrf/blob/master/samples/bluetooth/central_bas/src/main.c#L318-L366

Reply:
Alright, this is an example for a central. But do you also have an example for a peripheral?

But do you think it is a good idea to add this to the sample? I for myself like to have clean samples that show me one specific thing. In this example it is to create a Coded PHY connection. Adding 50+ lines of code increases the complexity of the sample without any benefit for the core idea. This may confuse people. But that's just my opinion :)

Reply:
It should be the same for peripheral, apart from who initiates it.

In your samples none of them initiates security, so these won't be used.
I don't think having pairing in this sample is needed, so feel free to exclude it for the sample. Makes sense to limit it to just connection setup on LE Coded PHY.